### PR TITLE
Rahu mongo ck

### PR DESCRIPTION
**Pull request by rahu-eGov**

### CODE REVIEW

File: domain-services/fiscal-event-service/src/main/resources/application.properties [hunk 1]
```diff
@@ -43,4 +43,6 @@ ifix.master.government.search.path=/government/v1/_search
 
 ifix.master.project.host=http://localhost:8030
 ifix.master.project.context.path=/ifix-master-data
-ifix.master.project.search.path=/project/v1/_search
\ No newline at end of file
+ifix.master.project.search.path=/project/v1/_search
+
+mongock.change-logs-scan-package=org.egov.migration
```
Comment (rushang7-eGov) [inline domain-services/fiscal-event-service/src/main/resources/application.properties:48]: Add extra line break

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package org.egov.migration;
```
Comment (rushang7-eGov) [inline domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java:1]: Rename the file

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        mongoCollection.createIndex(Indexes.ascending("eventType"));
+        mongoCollection.createIndex(Indexes.ascending("eventTime"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+        mongoCollection.createIndex(Indexes.ascending("referenceId"));
+    }
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914185000_createFiscalEventIndex", author = FE_INDEX_AUTHOR, order = "002")
```
Comment (rushang7-eGov) [inline domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java:36]: Combine in a single function

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+@ChangeLog(order = "001")
+@Slf4j
+public class MongoIndexScript {
+    public static final String DE_INDEX_AUTHOR = "iFIX-Department-Entity-Service";
+
+    /**
+     * @param db
+     */
+    @ChangeSet(id = "V20210914184200_createBasicFiscalEventIndex", author = DE_INDEX_AUTHOR, order = "001")
```
Comment (rushang7-eGov) [inline domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java:21]: Correct the name

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    public void createBasicDepartmentEntityIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("departmentEntity");
```
Comment (rushang7-eGov) [inline domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java:23]: Name the collection correctly

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+        mongoCollection.createIndex(Indexes.ascending("departmentId"));
+        mongoCollection.createIndex(Indexes.ascending("label"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+        mongoCollection.createIndex(Indexes.ascending("level"));
```
Comment (rushang7-eGov) [inline domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java:28]: Add index for "parent" attribute

File: domain-services/ifix-master-data-service/src/main/java/org/egov/migration/V20210915104900_MongoIndexScript.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+     */
+    @ChangeSet(id = "V20210914175400_createProjectIndex", author = MDS_INDEX_AUTHOR, order = "005")
+    public void createProjectIndex(MongoDatabase db) {
+        MongoCollection<Document> mongoCollection = db.getCollection("project");
+
+        mongoCollection.createIndex(Indexes.ascending("name"));
+        mongoCollection.createIndex(Indexes.ascending("code"));
+        mongoCollection.createIndex(Indexes.ascending("tenantId"));
+        mongoCollection.createIndex(Indexes.ascending("expenditureId"));
+        mongoCollection.createIndex(Indexes.ascending("departmentId"));
```
Comment (rushang7-eGov) [inline domain-services/ifix-master-data-service/src/main/java/org/egov/migration/V20210915104900_MongoIndexScript.java:84]: Add index for ancestry departmentEntityId.

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]

File: domain-services/fiscal-event-service/src/main/resources/application.properties (hunk 1 above)
Comment (rahu-eGov) [reply domain-services/fiscal-event-service/src/main/resources/application.properties:49]: done

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java (hunk 2 above)
Comment (rahu-eGov) [reply domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java]: done

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java (hunk 3 above)
Comment (rahu-eGov) [reply domain-services/fiscal-event-service/src/main/java/org/egov/migration/MongoIndexScript.java]: done

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java (hunk 4 above)
Comment (rahu-eGov) [reply domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java:21]: done

File: domain-services/ifix-master-data-service/src/main/java/org/egov/migration/V20210915104900_MongoIndexScript.java (hunk 7 above)
Comment (rahu-eGov) [reply domain-services/ifix-master-data-service/src/main/java/org/egov/migration/V20210915104900_MongoIndexScript.java]: done

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/V20210921123800_MongoIndexScript.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.code"));
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.name"));
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.ancestry.id"));
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.ancestry.code"));
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.ancestry.name"));
+        mongoCollection.createIndex(Indexes.ascending("departmentEntity.ancestry.hierarchyLevel"));
+        mongoCollection.createIndex(Indexes.ascending("expenditure.id"));
+        mongoCollection.createIndex(Indexes.ascending("project.id"));
+        mongoCollection.createIndex(Indexes.ascending("parentEventId"));
+        mongoCollection.createIndex(Indexes.ascending("amountDetails.coa.id"));
```
Comment (rushang7-eGov) [inline domain-services/fiscal-event-service/src/main/java/org/egov/migration/V20210921123800_MongoIndexScript.java:40]: "amountDetails.amount" ??

File: domain-services/fiscal-event-service/src/main/resources/application.properties (hunk 1 above)
Comment (rushang7-eGov) [reply domain-services/fiscal-event-service/src/main/resources/application.properties:49]: Add the extra line break at the END of the file. Not here.

File: domain-services/ifix-department-entity-service/src/main/resources/application.properties [hunk 9]
```diff
@@ -17,3 +17,5 @@ ifix.master.government.search.path=/government/v1/_search
 ifix.master.department.host=http://localhost:8030
 ifix.master.department.context.path=/ifix-master-data
 ifix.master.department.search.path=/department/v1/_search
+
+mongock.change-logs-scan-package=org.egov.migration
```
Comment (rushang7-eGov) [inline domain-services/ifix-department-entity-service/src/main/resources/application.properties:21]: Add line break at the end of the file

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java (hunk 5 above)
Comment (rahu-eGov) [reply domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java]: done

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]

File: domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java (hunk 6 above)
Comment (rahu-eGov) [reply domain-services/ifix-department-entity-service/src/main/java/org/egov/migration/MongoIndexScript.java]: done

File: domain-services/fiscal-event-service/src/main/java/org/egov/migration/V20210921123800_MongoIndexScript.java (hunk 8 above)
Comment (rahu-eGov) [reply domain-services/fiscal-event-service/src/main/java/org/egov/migration/V20210921123800_MongoIndexScript.java:40]: done

File: domain-services/ifix-department-entity-service/src/main/resources/application.properties (hunk 9 above)
Comment (rahu-eGov) [reply domain-services/ifix-department-entity-service/src/main/resources/application.properties:21]: done

**Review by rushang7-eGov — COMMENTED** [review]
Instructions to test on DEV environment:
1. Manually, Drop all the indexes from MongoDB.
2. Build and deploy new docker images of the services.
3. Check in MongoDB if indexes are getting created.
4. Verify the migration collection(table) created by Mongock. 

Once this is verified in DEV, I will approve and merge the PR.

**Pull request merged**